tutorials/optimization/cpu_optimization.rst

```diff
@@ -186,73 +193,82 @@ Third parties provide support for several other languages, including `Rust
+move further towards peak CPU efficiency.
 
 The disadvantage of threads is that you have to be incredibly careful. As each
 CPU core operates independently, they can end up trying to access the same
 memory at the same time. One thread can be reading to a variable while another
-is writing. Before you use threads make sure you understand the dangers and how
-to try and prevent these race conditions.
+is writing: this is called a *race condition*. Before you use threads, make sure
+you understand the dangers and how to try and prevent these race conditions.
+
+Threasd can also make debugging considerably more difficult. The GDScript
+debugger doesn't support setting up breakpoints in threads yet.
 
-For more information on threads see :ref:`doc_using_multiple_threads`.
+For more information on threads, see :ref:`doc_using_multiple_threads`.
 
 SceneTree
 =========
 
 Although Nodes are an incredibly powerful and versatile concept, be aware that
-every node has a cost. Built in functions such as `_process()` and
+every node has a cost. Built-in functions such as `_process()` and
 `_physics_process()` propagate through the tree. This housekeeping can reduce
-performance when you have very large numbers of nodes.
+performance when you have very large numbers of nodes (usually in the thousands).
 
-Each node is handled individually in the Godot renderer so sometimes a smaller
+Each node is handled individually in the Godot renderer. Therefore, a smaller
 number of nodes with more in each can lead to better performance.
 
 One quirk of the :ref:`SceneTree <class_SceneTree>` is that you can sometimes
-get much better performance by removing nodes from the SceneTree, rather than
-by pausing or hiding them. You don't have to delete a detached node. You
-can for example, keep a reference to a node, detach it from the scene tree, then
-reattach it later. This can be very useful for adding and removing areas from a
-game for example.
+get much better performance by removing nodes from the SceneTree, rather than by
+pausing or hiding them. You don't have to delete a detached node. You can for
+example, keep a reference to a node, detach it from the scene tree using
+:ref:`Node.remove_child(node) <class_Node_method_remove_child>`, then reattach
+it later using :ref:`Node.add_child(node) <class_Node_method_add_child>`.
+This can be very useful for adding and removing areas from a game, for example.
 
 You can avoid the SceneTree altogether by using Server APIs. For more
 information, see :ref:`doc_using_servers`.
 
 Physics
 =======
 
-In some situations physics can end up becoming a bottleneck, particularly with
-complex worlds, and large numbers of physics objects.
+In some situations, physics can end up becoming a bottleneck. This is
+particularly the case with complex worlds and large numbers of physics objects.
 
-Some techniques to speed up physics:
+Here are some techniques to speed up physics:
 
```

Comment:
Shouldnt it just be a single "_" at the end of the link? I may be wrong, feel free to ignore.